docs/6.x/storage.md

```diff
@@ -0,0 +1,346 @@
+class.
+
+Gravity clusters come with a pre-installed storage class called `openebs-hostpath`
+that uses `/var/lib/gravity/openebs/local` directory to store the persistent
+volumes data.
+
+!!! note "Custom state directory":
+    If cluster is using custom state directory location, say `/opt/gravity`,
+    the local volumes data will be stored under `/opt/gravity/openebs/local`
+    on host, but will still map to the `/var/lib/gravity/openebs/local` inside
+    the master (planet) container.
+
+To use the host volume storage, first define a persistent volume claim that
+uses the appropriate storage class, for example:
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: claim-hostpath
+spec:
+  storageClassName: openebs-hostpath
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+And then consume the claim in a pod template:
+
+```yaml
+volumes:
+- name: local-vol-from-hostpath
+  persistentVolumeClaim:
+    claimName: claim-hostpath
+```
+
+If you want to keep the local volumes data on a separate device, it is a common
+practice to mount it under `/var/lib/gravity/openebs/local` before initial
+cluster installation.
+
+Alternatively, you can also create another storage class that will use the
+host volume provisioner like described [OpenEBS documentation](https://docs.openebs.io/docs/next/localpv.html#how-to-use-openebs-local-pvs)
+and use it in your persistent volume claims instead of the default one.
+
+## Local Provisioner / Block Devices
+
+OpenEBS local provisioner also supports using block devices directly.
+
+To use a block device in a persistent volume, create a persistent volume claim
+that uses `openebs-device` storage class which is pre-created during cluster
+installation:
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: claim-device
+spec:
+  storageClassName: openebs-device
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+When persistent volume claim is attached to a pod, OpenEBS will find a matching
+block device among its managed devices pool (which can be viewed using
+`kubectl get blockdevices --all-namespaces`) and use it in a persistent volume:
+
+```yaml
+volumes:
+- name: local-vol-from-device
+  persistentVolumeClaim:
+    claimName: claim-device
+```
+
+See [OpenEBS documentation](https://docs.openebs.io/docs/next/localpv.html#how-to-use-openebs-local-pvs)
+on local provisioner to learn how to create another device-based storage class
+and tweak its parameters, if needed.
+
+## cStor Engine
+
+cStor is the most sophisticated storage engine supported by OpenEBS that provides
+features such as data replication, thin provisioning, snapshotting and so on.
+cStor serves block storage to containers over iSCSI interface.
+
+!!! note "iSCSI tools":
+    iSCSI tools must be installed on the cluster nodes in order for cStor volumes
```

Comment:
Just curious, why doesn't this work by just having the iscsi tools within planet??

Reply:
It does :) At least in my brief testing I think it worked. The reason I didn't include it in this "first round" of integration, is I think just adding iscsi tools to planet is not enough, it should be a bit smarter, for example, the planet should start iscsid conditionally, only if it's not already present on host, otherwise it will fail and this will lead to degraded status. I just wanted to focus on local pv initially, we can work on smoother cstor integration after that.